src/main/java/view_post/drivers/ViewPostGateway.java

```diff
@@ -1,14 +1,123 @@
 package view_post.drivers;
 
+import com.opencsv.CSVReader;
+import com.opencsv.exceptions.CsvException;
 import view_post.use_case.ViewPostDsGateway;
 
+import java.io.IOException;
+import java.io.Reader;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+
 public class ViewPostGateway implements ViewPostDsGateway {
+    // partialpath is a String that contains the partial directory that leads to all csv files in the program
+    // it is isolated from the file name so that in case the file is moved, the code is still compatible
+    private final String partialPath;
+
+    public ViewPostGateway(String partialPath){
+        this.partialPath = partialPath;
+    }
+
     /**
      * Return a list of strings representing the data of the post with matching post IDs.
      */
     public String[] getPostInfo(int postID) {
-        String[] postInfo = new String[5];
+        String[] postInfo = new String[9];
+
+        String[] rawPostData = this.getPostData(postID);
+        int userID = Integer.parseInt(rawPostData[1]);
+        String userEmail = this.getUserEmail(userID);
+
+        //poster email
+        postInfo[0] = userEmail;
+        //post body
+        postInfo[1] = rawPostData[3];
+        //post tags
+        postInfo[2] = rawPostData[4];
+        //replies
+        postInfo[3] = rawPostData[9];
+        //deadline
+        postInfo[4] = rawPostData[6];
+        //creation date
+        postInfo[5] = rawPostData[7];
+        //collaborators
+        postInfo[6] = rawPostData[5];
+        // post id
+        postInfo[7] = rawPostData[0];
+        //title
+        postInfo[8] = rawPostData[2];
 
         return postInfo;
     }
+    /**
+     * Retrieves email of a User from the database
+     * @param userid user's ID
+     * @return user's email
+     */
+    private String getUserEmail(int userid){
+        try {
+            // finding and retrieving the current user's data
+            List<String[]> allUsers = readAllLines(Paths.get(partialPath+"users.csv"));
```

Comment:
Skip the first line here. There will be an numberFormatException thrown here because it will try to do Integer.parseInt() on the header.